rpcserver.go

```diff
@@ -6743,6 +6747,38 @@ func (r *rpcServer) ListPermissions(_ context.Context,
 	}, nil
 }
 
+// CheckMacaroonPermissions checks the caveats and permissions of a macaroon.
+func (r *rpcServer) CheckMacaroonPermissions(ctx context.Context,
+	req *lnrpc.CheckMacPermRequest) (*lnrpc.CheckMacPermResponse, error) {
+
+	// Turn grpc macaroon permission into bakery.Op for the server to
+	// process.
+	permissions := make([]bakery.Op, 0)
+	for _, perm := range req.Permissions {
+		newPerm := bakery.Op{
+			Entity: perm.Entity,
+			Action: perm.Action,
+		}
+
+		permissions = append(permissions, newPerm)
+	}
+
+	err := r.macService.CheckMacAuth(
+		ctx, hex.EncodeToString(req.Macaroon), permissions,
+		"/lnrpc.Lightning/CheckMacaroonPermissions",
+	)
+	if err != nil {
+		resp := lnrpc.CheckMacPermResponse{
```

Comment:
Ah, I thought we would return the actual error here so the caller gets a bit more info _why_ the macaroon isn't valid. I see your point about returning the error message as part of the response gRPC message instead of returning the error as a failure on the connection level. I guess it is annoying on the user side of the API if you get an error and don't know if the connection failed, `lnd` has a problem or just the macaroon being invalid.

There is a discussion going on here https://github.com/lightningnetwork/lnd/pull/5436 and I guess we didn't come to a conclusion yet.

So to not block this PR on that discussion, what do you think about doing this here (instead of `Valid: false`):
```
return nil, status.Error(codes.InvalidArgument, err.Error())
```

That way you know on the client side it's not a general problem (e.g. `lnd` is down) because of the specific error code. But you still get a message that you can show to the user if necessary.

Reply:
Ahh I see what you're saying. Sounds good to me